cpp/include/cudf/functions.h

```diff
@@ -730,3 +730,20 @@ gdf_error gdf_from_dlpack(gdf_column** columns,
 gdf_error gdf_to_dlpack(DLManagedTensor_ *tensor,
                         gdf_column const * const * columns,
                         gdf_size_type num_columns);
+
+/* ----------------------------------------------------------------------------*/
+/** 
+ * @brief Checks if the rows of a `table` are sorted in a lexicographical order.
+ * 
+ * @param[in] input_table        whose rows needs to be validated 
+ * @param[in] descending         vector representing the relation between columns
+ *                               (0 is ascending order and 1 is descending)
+ * @param[in] nulls_are_smallest Flag to indicate if nulls are to be considered
+ * smaller than non-nulls or viceversa
+ * 
+ * @returns true - if sorted , false - if not.
+ */
+/* ----------------------------------------------------------------------------*/
+bool is_sorted(cudf::table const &table,
```

Comment:
functions.h will eventually go away as we move stuff to functionality-specific headers. So I think this belongs somewhere else. Specifically, I think it should be in a header called `include/cudf/predicates.hpp`. Also, it should be in `namespace cudf`.